### ISSUE TITLE
Objcpp include cpp header broken?

### ISSUE THREAD

**Issue by canatella** [dropbox/djinni#193]
Hello,

I'm updating my codebase against a more recent djinni and I stumble upon an issue.

IDL file:

    report_foo = record +c {
      field: string;
    }

Djinni command line:

    src/run --idl test.djinni --cpp-header-out tmp/cpp-headers --cpp-out tmp/cpp --cpp-include-prefix cpp-headers/ --objc-out tmp/objc --objcpp-out tmp/objcpp --objcpp-include-objc-prefix objc/ --objcpp-include-cpp-prefix cpp-headers/

Resulting tree:

    $ tree tmp
    tmp
    ├── cpp
    ├── cpp-headers
    │   └── report_foo_base.hpp
    ├── objc
    │   ├── ReportFoo.h
    │   └── ReportFoo.mm
    └── objcpp
        ├── ReportFoo+Private.h
        └── ReportFoo+Private.mm
    
    4 directories, 5 files

The tmp ReportFoo+Private.h file resulting:

    // AUTOGENERATED FILE - DO NOT MODIFY!
    // This file generated by Djinni from test.djinni
    
    #import "objc/ReportFoo.h"
    #include "cpp-headers/../report_foo.hpp"
    
    ...
    
    }  // namespace djinni_generated

The generated include

    #include "cpp-headers/../report_foo.hpp"

does not make sense to me. I'd put my handwritten cpp headers in a directory handwritten/cpp-headers/ and I expect the compiler would find all the headers with a -Ihandwritten -Itmp but with this includes, it's broken. It looks like this has been introduced in b38789e6f8b012beed09d9a27752d27b7404f29f.

Am I doing something wrong here ?

**Comment by canatella** [comment]
For information, With this patch, it's working ok:

```Diff
diff --git a/src/source/ObjcppGenerator.scala b/src/source/ObjcppGenerator.scala
index d28384c..d0b5e71 100644
--- a/src/source/ObjcppGenerator.scala
+++ b/src/source/ObjcppGenerator.scala
@@ -302,8 +302,8 @@ class ObjcppGenerator(spec: Spec) extends Generator(spec) {
     val noBaseSelf = objcMarshal.typename(ident, r) // Used for constant names
     val cppSelf = cppMarshal.fqTypename(ident, r)

-    refs.privHeader.add("!#import " + q(spec.objcppIncludeObjcPrefix + (if(r.ext.objc) "../" else "") + headerName(ident)))
-    refs.privHeader.add("!#include " + q(spec.objcppIncludeCppPrefix + (if(r.ext.cpp) "../" else "") + spec.cppFileIdentStyle(ident) + "." + spec.cppHeaderExt))
+    refs.privHeader.add("!#import " + q(spec.objcppIncludeObjcPrefix + headerName(ident)))
+    refs.privHeader.add("!#include " + q(spec.objcppIncludeCppPrefix + spec.cppFileIdentStyle(ident) + "." + spec.cppHeaderExt))

     refs.body.add("#include <cassert>")
     refs.body.add("!#import " + q(spec.objcppIncludePrefix + objcppMarshal.privateHeaderName(objcName)))
```

I don't know why these "../" where introduced but breaks for me here.

**Comment by guycnicholas** [comment]
Interesting timing...this is being discussed here: https://github.com/dropbox/djinni/pull/191

**Comment by artwyman** [comment]
Work on the latest version of a fix to this is now in #203.

**Comment by artwyman** [comment]
#203 is now merged
